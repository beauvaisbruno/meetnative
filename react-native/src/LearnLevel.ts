import lang from "./lang";

export class LearnLevel {
  static beginner = "beginner";
  static intermediate = "intermediate";
  static fluent = "fluent";
  static all = [
    LearnLevel.beginner,
    LearnLevel.intermediate,
    LearnLevel.fluent
  ];
  static lang(level: string): string {
    if (level === LearnLevel.beginner) return lang("Beginner A1/A2");
    if (level === LearnLevel.intermediate) return lang("Intermediate B1/B2");
    if (level === LearnLevel.fluent) return lang("Fluent C1/C2");
    return lang("Unknow");
  }
  static sortKeysByValue(langs: {
    [p: string]: string;
  }): { [p: string]: string } {
    const resObj: { [p: string]: string } = {};
    const keys = Object.keys(langs);
    keys.sort(
      (a: string, b: string): number => {
        if (langs[a] === langs[b]) return 0;
        if (langs[a] === LearnLevel.fluent) return -1;
        if (langs[a] === LearnLevel.beginner) return 1;
        if (langs[b] === LearnLevel.fluent) return 1;
        // if (langs[b] === TeachLevel.fluent)
        return -1;
      }
    );
    keys.forEach(key => {
      resObj[key] = langs[key];
    });
    return resObj;
  }
}
