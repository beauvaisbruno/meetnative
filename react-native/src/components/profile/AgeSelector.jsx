"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var native_base_1 = require("native-base");
var lang_1 = require("./lang");
var React = require("react");
var custom_1 = require("../../config/native-base-theme/variables/custom");
var createRows = function () {
    var rows = [];
    for (var i = 18; i < 77; i++) {
        rows.push(
        //@ts-ignore
        <native_base_1.Picker.Item label={lang_1.default("{age} yo", { age: i })} value={i} key={i}/>);
    }
    return rows;
};
exports.AgeSelector = function (_a) {
    var value = _a.value, onChange = _a.onChange;
    return (<native_base_1.Picker testID={"picker"} mode="dropdown" iosIcon={<native_base_1.Icon name="arrow-down"/>} placeholder={lang_1.default("Select your age")} placeholderStyle={{ color: custom_1.default.textGrey }} placeholderIconColor={custom_1.default.textGrey} style={{ width: undefined }} selectedValue={value} onValueChange={function (age) {
        onChange(age);
    }}>
      {createRows()}
    </native_base_1.Picker>);
};
