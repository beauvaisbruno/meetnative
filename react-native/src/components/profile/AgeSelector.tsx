import { Icon, Picker } from "native-base";
import lang from "./lang";
import * as React from "react";
import custom from "../../config/native-base-theme/variables/custom";

const createRows = () => {
  const rows = [];
  for (let i = 18; i < 77; i++) {
    rows.push(
      //@ts-ignore
      <Picker.Item label={lang("{age} yo", { age: i })} value={i} key={i} />
    );
  }
  return rows;
};

export const AgeSelector = ({
  value,
  onChange
}: {
  value: number;
  onChange: (age: number) => void;
}) => {
  return (
    <Picker
      testID={"picker"}
      mode="dropdown"
      iosIcon={<Icon name="arrow-down" />}
      placeholder={lang("Select your age")}
      placeholderStyle={{ color: custom.textGrey }}
      placeholderIconColor={custom.textGrey}
      style={{ width: undefined }}
      selectedValue={value}
      onValueChange={(age: number) => {
        onChange(age);
      }}
    >
      {createRows()}
    </Picker>
  );
};
