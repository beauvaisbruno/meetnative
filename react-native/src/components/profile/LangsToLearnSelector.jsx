"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var native_base_1 = require("native-base");
var React = require("react");
var lang_1 = require("./lang");
var Language_1 = require("../../Language");
var LearnLevel_1 = require("../../LearnLevel");
var react_native_responsive_screen_1 = require("react-native-responsive-screen");
var utilsUi_1 = require("../../utils/utilsUi");
function addRow(languagesToLearn) {
    for (var _i = 0, _a = Language_1.default.all; _i < _a.length; _i++) {
        var language = _a[_i];
        if (languagesToLearn[language] === undefined) {
            var notUsedLangObj = {};
            notUsedLangObj[language] = LearnLevel_1.LearnLevel.beginner;
            return notUsedLangObj;
        }
    }
    return {};
}
var createLevels = function () {
    var rows = [];
    LearnLevel_1.LearnLevel.all.forEach(function (level) {
        rows.push(<native_base_1.Picker.Item label={LearnLevel_1.LearnLevel.lang(level)} value={level} key={level}/>);
    });
    return rows;
};
exports.LangsToLearnSelector = function (_a) {
    var languagesToLearn = _a.languagesToLearn, onChange = _a.onChange;
    var rows = [];
    var firstRow = true;
    Object.keys(languagesToLearn).forEach(function (language) {
        rows.push(<native_base_1.View key={language} style={{ flexDirection: "row" }}>
        <native_base_1.View key={language} style={{ width: react_native_responsive_screen_1.widthPercentageToDP("35%") }}>
          <native_base_1.Picker testID={"lang-" + language} mode="dropdown" iosIcon={<native_base_1.Icon name="arrow-down"/>} 
        // style={{ width: undefined }}
        selectedValue={language} onValueChange={function (newLanguage) {
            var newLanguagesToLearn = __assign({}, languagesToLearn);
            newLanguagesToLearn[newLanguage] = languagesToLearn[language];
            delete newLanguagesToLearn[language];
            onChange(newLanguagesToLearn);
        }}>
            {utilsUi_1.createLangPickers()}
          </native_base_1.Picker>
        </native_base_1.View>
        <native_base_1.View key={languagesToLearn[language]} style={{ width: react_native_responsive_screen_1.widthPercentageToDP("50%") }}>
          <native_base_1.Picker testID={"level-" + language} mode="dropdown" iosIcon={<native_base_1.Icon name="arrow-down"/>} selectedValue={languagesToLearn[language]} onValueChange={function (newLevel) {
            var newLanguagesToLearn = __assign({}, languagesToLearn);
            newLanguagesToLearn[language] = newLevel;
            onChange(newLanguagesToLearn);
        }}>
            {createLevels()}
          </native_base_1.Picker>
        </native_base_1.View>
        {Object.keys(languagesToLearn).length > 1 && (<native_base_1.Button testID={"delete-" + language} transparent primary onPress={function () {
            var newLanguagesToLearn = __assign({}, languagesToLearn);
            delete newLanguagesToLearn[language];
            onChange(newLanguagesToLearn);
        }}>
            <native_base_1.Icon style={{
            paddingRight: 0,
            marginRight: 0,
            paddingLeft: 0,
            marginLeft: 0
        }} name="delete-forever" type="MaterialIcons" fontSize={30}/>
          </native_base_1.Button>)}
      </native_base_1.View>);
        firstRow = false;
    });
    return (<native_base_1.View>
      <native_base_1.View padder>
        <native_base_1.Text style={{ textDecorationLine: "underline" }}>
          {lang_1.default("Langage to learn")}
        </native_base_1.Text>
      </native_base_1.View>
      {rows}
      {Object.keys(languagesToLearn).length <= 2 && (<native_base_1.Button testID={"addBtn"} transparent primary onPress={function () {
        onChange(__assign({}, languagesToLearn, addRow(languagesToLearn)));
    }}>
          <native_base_1.Icon name="plus-circle" type="FontAwesome" fontSize={30}/>
          <native_base_1.Text>{lang_1.default("Add a langage")}</native_base_1.Text>
        </native_base_1.Button>)}
    </native_base_1.View>);
};
exports.default = exports.LangsToLearnSelector;
