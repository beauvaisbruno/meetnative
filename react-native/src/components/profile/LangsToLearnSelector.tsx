import { View, Text, Picker, Icon, Button } from "native-base";
import * as React from "react";
import lang from "./lang";
import Language from "../../Language";
import { LearnLevel } from "../../LearnLevel";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import { createLangPickers } from "../../utils/utilsUi";

function addRow(languagesToLearn: {
  [key: string]: string;
}): { [key: string]: string } {
  for (const language of Language.all) {
    if (languagesToLearn[language] === undefined) {
      const notUsedLangObj: { [key: string]: string } = {};
      notUsedLangObj[language] = LearnLevel.beginner;
      return notUsedLangObj;
    }
  }
  return {};
}

const createLevels = () => {
  const rows: JSX.Element[] = [];
  LearnLevel.all.forEach((level: string) => {
    rows.push(
      <Picker.Item label={LearnLevel.lang(level)} value={level} key={level} />
    );
  });
  return rows;
};

export const LangsToLearnSelector = ({
  languagesToLearn,
  onChange
}: {
  languagesToLearn: { [key: string]: string };
  onChange: (languagesToLearn: { [key: string]: string }) => void;
}) => {
  const rows: JSX.Element[] = [];
  let firstRow = true;
  Object.keys(languagesToLearn).forEach((language: string) => {
    rows.push(
      <View key={language} style={{ flexDirection: "row" }}>
        <View key={language} style={{ width: wp("35%") }}>
          <Picker
            testID={"lang-" + language}
            mode="dropdown"
            iosIcon={<Icon name="arrow-down" />}
            selectedValue={language}
            onValueChange={(newLanguage: string) => {
              const newLanguagesToLearn = { ...languagesToLearn };
              newLanguagesToLearn[newLanguage] = languagesToLearn[language];
              delete newLanguagesToLearn[language];
              onChange(newLanguagesToLearn);
            }}
          >
            {createLangPickers()}
          </Picker>
        </View>
        <View key={languagesToLearn[language]} style={{ width: wp("50%") }}>
          <Picker
            testID={"level-" + language}
            mode="dropdown"
            iosIcon={<Icon name="arrow-down" />}
            selectedValue={languagesToLearn[language]}
            onValueChange={(newLevel: string) => {
              const newLanguagesToLearn = { ...languagesToLearn };
              newLanguagesToLearn[language] = newLevel;
              onChange(newLanguagesToLearn);
            }}
          >
            {createLevels()}
          </Picker>
        </View>
        {Object.keys(languagesToLearn).length > 1 && (
          <Button
            testID={"delete-" + language}
            transparent
            primary
            onPress={() => {
              const newLanguagesToLearn = { ...languagesToLearn };
              delete newLanguagesToLearn[language];
              onChange(newLanguagesToLearn);
            }}
          >
            <Icon
              style={{
                paddingRight: 0,
                marginRight: 0,
                paddingLeft: 0,
                marginLeft: 0
              }}
              name="delete-forever"
              type="MaterialIcons"
              fontSize={30}
            />
          </Button>
        )}
      </View>
    );
    firstRow = false;
  });
  return (
    <View>
      <View padder>
        <Text style={{ textDecorationLine: "underline" }}>
          {lang("Langage to learn")}
        </Text>
      </View>
      {rows}
      {Object.keys(languagesToLearn).length <= 2 && (
        <Button
          testID={"addBtn"}
          transparent
          primary
          onPress={() => {
            onChange({ ...languagesToLearn, ...addRow(languagesToLearn) });
          }}
        >
          <Icon name="plus-circle" type="FontAwesome" fontSize={30} />
          <Text>{lang("Add a langage")}</Text>
        </Button>
      )}
    </View>
  );
};

export default LangsToLearnSelector;
