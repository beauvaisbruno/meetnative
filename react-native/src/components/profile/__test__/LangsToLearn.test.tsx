import { shallow } from "enzyme";
import { AgeSelector } from "../AgeSelector";
import * as React from "react";
import toJson from "enzyme-to-json";
import LangsToLearnSelector from "../LangsToLearnSelector";
import { LearnLevel } from "../../../LearnLevel";

it("LangsToLearnSelector", async () => {
  const onChange = jest.fn();
  const tree = shallow(
    <LangsToLearnSelector
      languagesToLearn={{ spanish: LearnLevel.beginner }}
      onChange={onChange}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
  const addBtn = tree.findWhere(node => node.prop("testID") === "addBtn");
  addBtn.props().onPress();
  expect(onChange).toBeCalledWith({
    spanish: LearnLevel.beginner,
    english: LearnLevel.beginner
  });
  const tree2 = shallow(
    <LangsToLearnSelector
      languagesToLearn={{
        english: LearnLevel.beginner,
        spanish: LearnLevel.beginner,
        italian: LearnLevel.beginner
      }}
      onChange={onChange}
    />
  );
  expect(toJson(tree2)).toMatchSnapshot();
  const langPicker = tree2.findWhere(
    node => node.prop("testID") === "lang-english"
  );
  langPicker.props().onValueChange("spanish");
  expect(onChange).toBeCalledWith({
    spanish: LearnLevel.beginner,
    italian: LearnLevel.beginner
  });
  const levelPicker = tree2.findWhere(
    node => node.prop("testID") === "level-english"
  );
  levelPicker.props().onValueChange(LearnLevel.fluent);
  expect(onChange).toBeCalledWith({
    english: LearnLevel.fluent,
    spanish: LearnLevel.beginner,
    italian: LearnLevel.beginner
  });
  const deleteBtn = tree2.findWhere(
    node => node.prop("testID") === "delete-english"
  );
  deleteBtn.props().onPress();
  expect(onChange).toBeCalledWith({
    spanish: LearnLevel.beginner,
    italian: LearnLevel.beginner
  });
});
