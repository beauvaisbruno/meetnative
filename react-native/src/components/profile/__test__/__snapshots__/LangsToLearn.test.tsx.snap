// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangsToLearnSelector 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      Langage to learn
    </Styled(Text)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="spanish"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Styled(ViewNB)
      key="spanish"
      style={
        Object {
          "width": 262.5,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="spanish"
        testID="lang-spanish"
      >
        <Item
          key="arabic"
          label="Arabic"
          value="arabic"
        />
        <Item
          key="basque"
          label="Basque"
          value="basque"
        />
        <Item
          key="bulgarian"
          label="Bulgarian"
          value="bulgarian"
        />
        <Item
          key="catalan"
          label="Catalan"
          value="catalan"
        />
        <Item
          key="chinese"
          label="Chinese"
          value="chinese"
        />
        <Item
          key="croatian"
          label="Croatian"
          value="croatian"
        />
        <Item
          key="czech"
          label="Czech"
          value="czech"
        />
        <Item
          key="danish"
          label="Danish"
          value="danish"
        />
        <Item
          key="dutch"
          label="Dutch"
          value="dutch"
        />
        <Item
          key="english"
          label="English"
          value="english"
        />
        <Item
          key="estonian"
          label="Estonian"
          value="estonian"
        />
        <Item
          key="finnish"
          label="Finnish"
          value="finnish"
        />
        <Item
          key="french"
          label="French"
          value="french"
        />
        <Item
          key="galician"
          label="Galician"
          value="galician"
        />
        <Item
          key="german"
          label="German"
          value="german"
        />
        <Item
          key="greek"
          label="Greek"
          value="greek"
        />
        <Item
          key="hindi"
          label="Hindi"
          value="hindi"
        />
        <Item
          key="hungarian"
          label="Hungarian"
          value="hungarian"
        />
        <Item
          key="irish_gaelic"
          label="Irish Gaelic"
          value="irish_gaelic"
        />
        <Item
          key="italian"
          label="Italian"
          value="italian"
        />
        <Item
          key="japanese"
          label="Japanese"
          value="japanese"
        />
        <Item
          key="korean"
          label="Korean"
          value="korean"
        />
        <Item
          key="latvian"
          label="Latvian"
          value="latvian"
        />
        <Item
          key="lithuanian"
          label="Lithuanian"
          value="lithuanian"
        />
        <Item
          key="luxembourgish"
          label="Luxembourgish"
          value="luxembourgish"
        />
        <Item
          key="maltese"
          label="Maltese"
          value="maltese"
        />
        <Item
          key="polish"
          label="Polish"
          value="polish"
        />
        <Item
          key="portuguese"
          label="Portuguese"
          value="portuguese"
        />
        <Item
          key="romanian"
          label="Romanian"
          value="romanian"
        />
        <Item
          key="russian"
          label="Russian"
          value="russian"
        />
        <Item
          key="scottish_gaelic"
          label="Scottish Gaelic"
          value="scottish_gaelic"
        />
        <Item
          key="slovak"
          label="Slovak"
          value="slovak"
        />
        <Item
          key="slovenian"
          label="Slovenian"
          value="slovenian"
        />
        <Item
          key="spanish"
          label="Spanish"
          value="spanish"
        />
        <Item
          key="swedish"
          label="Swedish"
          value="swedish"
        />
        <Item
          key="turkish"
          label="Turkish"
          value="turkish"
        />
        <Item
          key="urdu"
          label="Urdu"
          value="urdu"
        />
        <Item
          key="welsh"
          label="Welsh"
          value="welsh"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      key="beginner"
      style={
        Object {
          "width": 375,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="beginner"
        testID="level-spanish"
      >
        <Item
          key="beginner"
          label="Beginner A1/A2"
          value="beginner"
        />
        <Item
          key="intermediate"
          label="Intermediate B1/B2"
          value="intermediate"
        />
        <Item
          key="fluent"
          label="Fluent C1/C2"
          value="fluent"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
  <Styled(Button)
    onPress={[Function]}
    primary={true}
    testID="addBtn"
    transparent={true}
  >
    <Styled(Icon)
      fontSize={30}
      name="plus-circle"
      type="FontAwesome"
    />
    <Styled(Text)>
      Add a langage
    </Styled(Text)>
  </Styled(Button)>
</Styled(ViewNB)>
`;

exports[`LangsToLearnSelector 2`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
  >
    <Styled(Text)
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      Langage to learn
    </Styled(Text)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="english"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Styled(ViewNB)
      key="english"
      style={
        Object {
          "width": 262.5,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="english"
        testID="lang-english"
      >
        <Item
          key="arabic"
          label="Arabic"
          value="arabic"
        />
        <Item
          key="basque"
          label="Basque"
          value="basque"
        />
        <Item
          key="bulgarian"
          label="Bulgarian"
          value="bulgarian"
        />
        <Item
          key="catalan"
          label="Catalan"
          value="catalan"
        />
        <Item
          key="chinese"
          label="Chinese"
          value="chinese"
        />
        <Item
          key="croatian"
          label="Croatian"
          value="croatian"
        />
        <Item
          key="czech"
          label="Czech"
          value="czech"
        />
        <Item
          key="danish"
          label="Danish"
          value="danish"
        />
        <Item
          key="dutch"
          label="Dutch"
          value="dutch"
        />
        <Item
          key="english"
          label="English"
          value="english"
        />
        <Item
          key="estonian"
          label="Estonian"
          value="estonian"
        />
        <Item
          key="finnish"
          label="Finnish"
          value="finnish"
        />
        <Item
          key="french"
          label="French"
          value="french"
        />
        <Item
          key="galician"
          label="Galician"
          value="galician"
        />
        <Item
          key="german"
          label="German"
          value="german"
        />
        <Item
          key="greek"
          label="Greek"
          value="greek"
        />
        <Item
          key="hindi"
          label="Hindi"
          value="hindi"
        />
        <Item
          key="hungarian"
          label="Hungarian"
          value="hungarian"
        />
        <Item
          key="irish_gaelic"
          label="Irish Gaelic"
          value="irish_gaelic"
        />
        <Item
          key="italian"
          label="Italian"
          value="italian"
        />
        <Item
          key="japanese"
          label="Japanese"
          value="japanese"
        />
        <Item
          key="korean"
          label="Korean"
          value="korean"
        />
        <Item
          key="latvian"
          label="Latvian"
          value="latvian"
        />
        <Item
          key="lithuanian"
          label="Lithuanian"
          value="lithuanian"
        />
        <Item
          key="luxembourgish"
          label="Luxembourgish"
          value="luxembourgish"
        />
        <Item
          key="maltese"
          label="Maltese"
          value="maltese"
        />
        <Item
          key="polish"
          label="Polish"
          value="polish"
        />
        <Item
          key="portuguese"
          label="Portuguese"
          value="portuguese"
        />
        <Item
          key="romanian"
          label="Romanian"
          value="romanian"
        />
        <Item
          key="russian"
          label="Russian"
          value="russian"
        />
        <Item
          key="scottish_gaelic"
          label="Scottish Gaelic"
          value="scottish_gaelic"
        />
        <Item
          key="slovak"
          label="Slovak"
          value="slovak"
        />
        <Item
          key="slovenian"
          label="Slovenian"
          value="slovenian"
        />
        <Item
          key="spanish"
          label="Spanish"
          value="spanish"
        />
        <Item
          key="swedish"
          label="Swedish"
          value="swedish"
        />
        <Item
          key="turkish"
          label="Turkish"
          value="turkish"
        />
        <Item
          key="urdu"
          label="Urdu"
          value="urdu"
        />
        <Item
          key="welsh"
          label="Welsh"
          value="welsh"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      key="beginner"
      style={
        Object {
          "width": 375,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="beginner"
        testID="level-english"
      >
        <Item
          key="beginner"
          label="Beginner A1/A2"
          value="beginner"
        />
        <Item
          key="intermediate"
          label="Intermediate B1/B2"
          value="intermediate"
        />
        <Item
          key="fluent"
          label="Fluent C1/C2"
          value="fluent"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
    <Styled(Button)
      onPress={[Function]}
      primary={true}
      testID="delete-english"
      transparent={true}
    >
      <Styled(Icon)
        fontSize={30}
        name="delete-forever"
        style={
          Object {
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
          }
        }
        type="MaterialIcons"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="spanish"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Styled(ViewNB)
      key="spanish"
      style={
        Object {
          "width": 262.5,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="spanish"
        testID="lang-spanish"
      >
        <Item
          key="arabic"
          label="Arabic"
          value="arabic"
        />
        <Item
          key="basque"
          label="Basque"
          value="basque"
        />
        <Item
          key="bulgarian"
          label="Bulgarian"
          value="bulgarian"
        />
        <Item
          key="catalan"
          label="Catalan"
          value="catalan"
        />
        <Item
          key="chinese"
          label="Chinese"
          value="chinese"
        />
        <Item
          key="croatian"
          label="Croatian"
          value="croatian"
        />
        <Item
          key="czech"
          label="Czech"
          value="czech"
        />
        <Item
          key="danish"
          label="Danish"
          value="danish"
        />
        <Item
          key="dutch"
          label="Dutch"
          value="dutch"
        />
        <Item
          key="english"
          label="English"
          value="english"
        />
        <Item
          key="estonian"
          label="Estonian"
          value="estonian"
        />
        <Item
          key="finnish"
          label="Finnish"
          value="finnish"
        />
        <Item
          key="french"
          label="French"
          value="french"
        />
        <Item
          key="galician"
          label="Galician"
          value="galician"
        />
        <Item
          key="german"
          label="German"
          value="german"
        />
        <Item
          key="greek"
          label="Greek"
          value="greek"
        />
        <Item
          key="hindi"
          label="Hindi"
          value="hindi"
        />
        <Item
          key="hungarian"
          label="Hungarian"
          value="hungarian"
        />
        <Item
          key="irish_gaelic"
          label="Irish Gaelic"
          value="irish_gaelic"
        />
        <Item
          key="italian"
          label="Italian"
          value="italian"
        />
        <Item
          key="japanese"
          label="Japanese"
          value="japanese"
        />
        <Item
          key="korean"
          label="Korean"
          value="korean"
        />
        <Item
          key="latvian"
          label="Latvian"
          value="latvian"
        />
        <Item
          key="lithuanian"
          label="Lithuanian"
          value="lithuanian"
        />
        <Item
          key="luxembourgish"
          label="Luxembourgish"
          value="luxembourgish"
        />
        <Item
          key="maltese"
          label="Maltese"
          value="maltese"
        />
        <Item
          key="polish"
          label="Polish"
          value="polish"
        />
        <Item
          key="portuguese"
          label="Portuguese"
          value="portuguese"
        />
        <Item
          key="romanian"
          label="Romanian"
          value="romanian"
        />
        <Item
          key="russian"
          label="Russian"
          value="russian"
        />
        <Item
          key="scottish_gaelic"
          label="Scottish Gaelic"
          value="scottish_gaelic"
        />
        <Item
          key="slovak"
          label="Slovak"
          value="slovak"
        />
        <Item
          key="slovenian"
          label="Slovenian"
          value="slovenian"
        />
        <Item
          key="spanish"
          label="Spanish"
          value="spanish"
        />
        <Item
          key="swedish"
          label="Swedish"
          value="swedish"
        />
        <Item
          key="turkish"
          label="Turkish"
          value="turkish"
        />
        <Item
          key="urdu"
          label="Urdu"
          value="urdu"
        />
        <Item
          key="welsh"
          label="Welsh"
          value="welsh"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      key="beginner"
      style={
        Object {
          "width": 375,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="beginner"
        testID="level-spanish"
      >
        <Item
          key="beginner"
          label="Beginner A1/A2"
          value="beginner"
        />
        <Item
          key="intermediate"
          label="Intermediate B1/B2"
          value="intermediate"
        />
        <Item
          key="fluent"
          label="Fluent C1/C2"
          value="fluent"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
    <Styled(Button)
      onPress={[Function]}
      primary={true}
      testID="delete-spanish"
      transparent={true}
    >
      <Styled(Icon)
        fontSize={30}
        name="delete-forever"
        style={
          Object {
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
          }
        }
        type="MaterialIcons"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="italian"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Styled(ViewNB)
      key="italian"
      style={
        Object {
          "width": 262.5,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="italian"
        testID="lang-italian"
      >
        <Item
          key="arabic"
          label="Arabic"
          value="arabic"
        />
        <Item
          key="basque"
          label="Basque"
          value="basque"
        />
        <Item
          key="bulgarian"
          label="Bulgarian"
          value="bulgarian"
        />
        <Item
          key="catalan"
          label="Catalan"
          value="catalan"
        />
        <Item
          key="chinese"
          label="Chinese"
          value="chinese"
        />
        <Item
          key="croatian"
          label="Croatian"
          value="croatian"
        />
        <Item
          key="czech"
          label="Czech"
          value="czech"
        />
        <Item
          key="danish"
          label="Danish"
          value="danish"
        />
        <Item
          key="dutch"
          label="Dutch"
          value="dutch"
        />
        <Item
          key="english"
          label="English"
          value="english"
        />
        <Item
          key="estonian"
          label="Estonian"
          value="estonian"
        />
        <Item
          key="finnish"
          label="Finnish"
          value="finnish"
        />
        <Item
          key="french"
          label="French"
          value="french"
        />
        <Item
          key="galician"
          label="Galician"
          value="galician"
        />
        <Item
          key="german"
          label="German"
          value="german"
        />
        <Item
          key="greek"
          label="Greek"
          value="greek"
        />
        <Item
          key="hindi"
          label="Hindi"
          value="hindi"
        />
        <Item
          key="hungarian"
          label="Hungarian"
          value="hungarian"
        />
        <Item
          key="irish_gaelic"
          label="Irish Gaelic"
          value="irish_gaelic"
        />
        <Item
          key="italian"
          label="Italian"
          value="italian"
        />
        <Item
          key="japanese"
          label="Japanese"
          value="japanese"
        />
        <Item
          key="korean"
          label="Korean"
          value="korean"
        />
        <Item
          key="latvian"
          label="Latvian"
          value="latvian"
        />
        <Item
          key="lithuanian"
          label="Lithuanian"
          value="lithuanian"
        />
        <Item
          key="luxembourgish"
          label="Luxembourgish"
          value="luxembourgish"
        />
        <Item
          key="maltese"
          label="Maltese"
          value="maltese"
        />
        <Item
          key="polish"
          label="Polish"
          value="polish"
        />
        <Item
          key="portuguese"
          label="Portuguese"
          value="portuguese"
        />
        <Item
          key="romanian"
          label="Romanian"
          value="romanian"
        />
        <Item
          key="russian"
          label="Russian"
          value="russian"
        />
        <Item
          key="scottish_gaelic"
          label="Scottish Gaelic"
          value="scottish_gaelic"
        />
        <Item
          key="slovak"
          label="Slovak"
          value="slovak"
        />
        <Item
          key="slovenian"
          label="Slovenian"
          value="slovenian"
        />
        <Item
          key="spanish"
          label="Spanish"
          value="spanish"
        />
        <Item
          key="swedish"
          label="Swedish"
          value="swedish"
        />
        <Item
          key="turkish"
          label="Turkish"
          value="turkish"
        />
        <Item
          key="urdu"
          label="Urdu"
          value="urdu"
        />
        <Item
          key="welsh"
          label="Welsh"
          value="welsh"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      key="beginner"
      style={
        Object {
          "width": 375,
        }
      }
    >
      <Styled(PickerNB)
        iosIcon={
          <Styled(Icon)
            name="arrow-down"
          />
        }
        mode="dropdown"
        onValueChange={[Function]}
        selectedValue="beginner"
        testID="level-italian"
      >
        <Item
          key="beginner"
          label="Beginner A1/A2"
          value="beginner"
        />
        <Item
          key="intermediate"
          label="Intermediate B1/B2"
          value="intermediate"
        />
        <Item
          key="fluent"
          label="Fluent C1/C2"
          value="fluent"
        />
      </Styled(PickerNB)>
    </Styled(ViewNB)>
    <Styled(Button)
      onPress={[Function]}
      primary={true}
      testID="delete-italian"
      transparent={true}
    >
      <Styled(Icon)
        fontSize={30}
        name="delete-forever"
        style={
          Object {
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
          }
        }
        type="MaterialIcons"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
